pkg/cloudprovider/vsphere/nodemanager.go

```diff
@@ -113,6 +114,10 @@ func (nm *NodeManager) shakeOutNodeIDLookup(ctx context.Context, nodeID string,
 			return vmDI, err
 		}
 
+		if err != vclib.ErrNoVMFound {
```

Comment:
This will short circuit the lookup by IP because if the hostname is overridden to be the IP, you will definitely fail the first with no VM found. I think if you remove this, you will see get the desired results.

Reply:
Not sure I follow, this only short circuits if the err is not `vclib.ErrNoVMFound`. So if the name check above fails with `vclib.ErrNoVMFound`, we would still run the FindVMByIP below right?

Reply:
Sorry my bad... that's right!